[plugin-web-app] Introduce expression to decode Data URLs (#3520)
vividus-framework/vividus: docs/modules/plugins/pages/plugin-web-app.adoc

include::partial$proxy-meta-tags.adoc[]

> == Expressions

Reviewer: ```suggestion
== xref:ROOT:glossary.adoc#_expression[Expressions]
```

NOTE: The expression parameters marked with *bold* are mandatory.

=== `decodeDataUrl`

> Expression parses https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs[Data URLs] and decodes its data if https://developer.mozilla.org/en-US/docs/Glossary/Base64[Base64] encoding is set. The result of the expression depends on the Data URL's https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types[MIME type]. For the binary MIME type the result will be a binary data and for the text it will be a string.

Reviewer: ```suggestion
Parses https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs[Data URLs] and decodes their data if https://developer.mozilla.org/en-US/docs/Glossary/Base64[Base64] encoding is set. The result of the expression depends on the Data URL's https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types[MIME type]. The result will be a binary data in case of the binary MIME type and it will be a string for the text MIME type.
```

> Parses https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs[Data URLs] and decodes their data if https://developer.mozilla.org/en-US/docs/Glossary/Base64[Base64] encoding is set. The result of the expression depends on the Data URL's https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types[MIME type]. The result will be a binary data in case of the binary MIME type and it will be a string for the text MIME type.

Reviewer: ```suggestion
Parses https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs[Data URL-s] and decodes their data if https://developer.mozilla.org/en-US/docs/Glossary/Base64[Base64] encoding is set. The result of the expression depends on the https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types[MIME type] of Data URL. The result will be a binary data in case of the binary MIME type and it will be a string for the text MIME type.
```

> WARNING: If expression is used with not a valid Data URL the exception will be thrown.

Reviewer: ```suggestion
WARNING: If expression is applied to invalid Data URL the error will be thrown.
```

[source, subs="+quotes"]
----
#{decodeDataUrl(*$dataUrl*)}
----

* *`$dataUrl`* - The Data URL to decode.

.Decoding inlined image
[source,gherkin]
----

> When I save `src` attribute value of element located `id(image)` to scopes scenario `image`

Reviewer: ```suggestion
When I save `src` attribute value of element located `By.id(image)` to scenario variable `image`
```
